tests/test_app.py

```diff
@@ -31,14 +32,39 @@ def auth_headers(workspace="workspace", user="user", expiry=None, scope="view"):
 
 def test_validate_invalid_token():
     url = "/workspace/workspace/current"
-    token = auth_token(url)
+    signer = itsdangerous.Signer("badsecret")
+    token = create_raw_token(
+        dict(
+            url=urljoin(client.base_url, "/workspace/workspace"),
+            user="user",
+            expiry=datetime.now(timezone.utc) - timedelta(hours=1),
```

Comment:
Should this be a valid expiry so we're only checking the `badsecret` portion?

Reply:
great catch!